qgis fails to compile using Qt 6.10.1

### What is the bug or the crash?

there appears an specific compiling error in [qgssnappingutils.cpp:649](https://github.com/qgis/QGIS/blob/a1764d651f9913521fe319d94d50678686791f4f/src/core/qgssnappingutils.cpp#L649) upon calling `QString::arg(const QgsPointLocator::Types&)`

```
FAILED: [code=1] src/core/CMakeFiles/qgis_core.dir/qgssnappingutils.cpp.o 
/usr/bin/x86_64-pc-linux-gnu-g++ -DCMAKE_SOURCE_DIR=\"/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3\" -DPROTOBUF_USE_DLLS -DQT_CONCURRENT_LIB -DQT_CORE5COMPAT_LIB -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_NO_DEPRECATED_WARNINGS -DQT_NO_FOREACH -DQT_POSITIONING_LIB -DQT_PRINTSUPPORT_LIB -DQT_SERIALPORT_LIB -DQT_SQL_LIB -DQT_SVG_LIB -DQT_USE_QSTRINGBUILDER -DQT_WIDGETS_LIB -DQT_XML_LIB -DSIP_VERSION=0x060e00 -DTEST_DATA_DIR=\"/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/tests/testdata\" -DWITH_COPC -DWITH_EPT -D_HAVE_PTHREAD_ -Dqgis_core_EXPORTS -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3_build/src/core/qgis_core_autogen/include -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3_build -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/external/poly2tri -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/providers/ept -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/providers/copc -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/providers/vpc -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3_build/src/core -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/external/meshOptimizer -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/3d -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/actions -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/annotations -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/auth -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/browser -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/callouts -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/classification -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/diagram -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/dxf -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/editform -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/effects -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/elevation -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/expression -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/externalstorage -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/fieldformatter -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/geometry -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/geocoding -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/gps -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/labeling -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/labeling/rules -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/layertree -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/layout -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/locator -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/maprenderer -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/mesh -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/metadata -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/network -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/numericformats -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/painting -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/pal -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/pdf -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/plot -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/pointcloud -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/pointcloud/expression -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/processing -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/processing/models -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/proj -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/project -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/providers -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/providers/arcgis -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/providers/memory -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/providers/gdal -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/providers/ogr -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/providers/meshmemory -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/providers/sensorthings -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/raster -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/renderer -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/scalebar -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/settings -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/sensor -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/stac -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/symbology -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/textrenderer -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/tiledscene -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/validity -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/vector -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/vectortile -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/web -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/external -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/external/delaunator-cpp -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/external/kdbush/include -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/external/nmea -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/external/rtree/include -I/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/external/tinygltf -isystem /var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/external/opencl-clhpp/include -isystem /usr/include/Qca-qt6/QtCrypto -isystem /usr/include/qt6/QtCore -isystem /usr/include/qt6 -isystem /usr/lib64/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtGui -isystem /usr/include/qt6/QtXml -isystem /usr/include/qt6/QtWidgets -isystem /usr/include/qt6/QtSvg -isystem /usr/include/qt6/QtNetwork -isystem /usr/include/qt6/QtSql -isystem /usr/include/qt6/QtConcurrent -isystem /usr/include/qt6/QtPositioning -isystem /usr/include/geos -isystem /usr/include/gdal -isystem /usr/include/qt6/QtDBus -isystem /usr/include/qt6/QtPrintSupport -isystem /usr/include/qt6/QtCore5Compat -isystem /usr/include/qt6/QtSerialPort  -march=native -O2 -pipe -fno-strict-aliasing -std=gnu++17 -fPIC -fvisibility=hidden -mno-direct-extern-access -MD -MT src/core/CMakeFiles/qgis_core.dir/qgssnappingutils.cpp.o -MF src/core/CMakeFiles/qgis_core.dir/qgssnappingutils.cpp.o.d -o src/core/CMakeFiles/qgis_core.dir/qgssnappingutils.cpp.o -c /var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/qgssnappingutils.cpp
/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/qgssnappingutils.cpp: In member function ‘QString QgsSnappingUtils::dump()’:
/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/qgssnappingutils.cpp:649:16: error: no matching function for call to ‘QString::arg(const QgsPointLocator::Types&)’
  646 |     msg += QString( "layer : %1\n"
      |            ~~~~~~~~~~~~~~~~~~~~~~~
  647 |                     "config: %2   tolerance %3 %4\n" )
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  648 |            .arg( layer.layer->name() )
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~
  649 |            .arg( layer.type ).arg( layer.tolerance ).arg( static_cast<int>( layer.unit ) );
      |            ~~~~^~~~~~~~~~~~~~
In file included from /usr/include/qt6/QtCore/qhashfunctions.h:9,
                 from /usr/include/qt6/QtCore/qlist.h:11,
                 from /usr/include/qt6/QtCore/qstringlist.h:6,
                 from /usr/include/qt6/QtGui/qcolor.h:10,
                 from /usr/include/qt6/QtGui/QColor:1,
                 from /var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/qgsmapsettings.h:21,
                 from /var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/qgssnappingutils.h:22,
                 from /var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/qgssnappingutils.cpp:16:
/usr/include/qt6/QtCore/qstring.h:324:27: note: candidate: ‘template<class T, typename std::enable_if<conjunction_v<std::disjunction<std::is_integral<_Tp>, std::conjunction<std::is_enum<_Tp>, std::negation<q23::is_scoped_enum<E, is_enum_v<E> > > > >, std::negation<std::disjunction<std::is_same<typename std::remove_cv<typename std::remove_reference<_Tp>::type>::type, _Float16>, std::is_same<typename std::remove_cv<typename std::remove_reference<_Tp>::type>::type, qfloat16>, std::is_floating_point<_Tp> > >, std::negation<std::conjunction<std::negation<QtPrivate::treat_as_integral_arg<typename std::remove_cv< <template-parameter-1-1> >::type> >, std::is_convertible<T, QAnyStringView> > > >, bool>::type <anonymous> > QString QString::arg(T, int, int, QChar) const’
  324 |     [[nodiscard]] QString arg(T a, int fieldWidth = 0, int base = 10,
      |                           ^~~
/usr/include/qt6/QtCore/qstring.h:324:27: note:   template argument deduction/substitution failed:
/usr/include/qt6/QtCore/qstring.h:323:53: error: no type named ‘type’ in ‘struct std::enable_if<false, bool>’
  323 |     template <typename T, if_integral_non_char<T> = true>
      |                                                     ^~~~
/usr/include/qt6/QtCore/qstring.h:339:27: note: candidate: ‘template<class T, typename std::enable_if<std::disjunction<std::is_same<typename std::remove_cv<typename std::remove_reference<_Tp>::type>::type, _Float16>, std::is_same<typename std::remove_cv<typename std::remove_reference<_Tp>::type>::type, qfloat16>, std::is_floating_point<_Tp> >::value, bool>::type <anonymous> > QString QString::arg(T, int, char, int, QChar) const’
  339 |     [[nodiscard]] QString arg(T a, int fieldWidth = 0, char format = 'g', int precision = -1,
      |                           ^~~
/usr/include/qt6/QtCore/qstring.h:339:27: note:   template argument deduction/substitution failed:
/usr/include/qt6/QtCore/qstring.h:338:50: error: no type named ‘type’ in ‘struct std::enable_if<false, bool>’
  338 |     template <typename T, if_floating_point<T> = true>
      |                                                  ^~~~
/usr/include/qt6/QtCore/qstring.h:344:27: note: candidate: ‘template<class T, typename std::enable_if<std::conjunction<std::negation<QtPrivate::treat_as_integral_arg<typename std::remove_cv< <template-parameter-1-1> >::type> >, std::is_convertible<T, QAnyStringView> >::value, bool>::type <anonymous> > QString QString::arg(const T&, int, QChar) const’
  344 |     [[nodiscard]] QString arg(const T &a, int fieldWidth = 0, QChar fillChar = u' ') const
      |                           ^~~
/usr/include/qt6/QtCore/qstring.h:344:27: note:   template argument deduction/substitution failed:
/usr/include/qt6/QtCore/qstring.h:343:47: error: no type named ‘type’ in ‘struct std::enable_if<false, bool>’
  343 |     template <typename T, if_string_like<T> = true>
      |                                               ^~~~
/usr/include/qt6/QtCore/qstring.h:364:5: note: candidate: ‘template<class ... Args> typename std::enable_if<((sizeof... (Args) >= 2) && conjunction_v<std::conjunction<std::negation<QtPrivate::treat_as_integral_arg<typename std::remove_cv<Args>::type> >, std::is_convertible<Args, QAnyStringView> >...>), QString>::type QString::arg(Args&& ...) const’
  364 |     arg(Args &&...args) const
      |     ^~~
/usr/include/qt6/QtCore/qstring.h:364:5: note:   template argument deduction/substitution failed:
/usr/include/qt6/QtCore/qstring.h: In substitution of ‘template<class ... Args> typename std::enable_if<((sizeof... (Args) >= 2) && conjunction_v<std::conjunction<std::negation<QtPrivate::treat_as_integral_arg<typename std::remove_cv<Args>::type> >, std::is_convertible<Args, QAnyStringView> >...>), QString>::type QString::arg(Args&& ...) const [with Args = {const QFlags<QgsPointLocator::Type>&}]’:
/var/tmp/portage/sci-geosciences/qgis-3.44.3/work/qgis-3.44.3/src/core/qgssnappingutils.cpp:649:16:   required from here
  646 |     msg += QString( "layer : %1\n"
      |            ~~~~~~~~~~~~~~~~~~~~~~~
  647 |                     "config: %2   tolerance %3 %4\n" )
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  648 |            .arg( layer.layer->name() )
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~
  649 |            .arg( layer.type ).arg( layer.tolerance ).arg( static_cast<int>( layer.unit ) );
      |            ~~~~^~~~~~~~~~~~~~
/usr/include/qt6/QtCore/qstring.h:364:5: error: no type named ‘type’ in ‘struct std::enable_if<false, QString>’
  364 |     arg(Args &&...args) const
      |     ^~~
```

### Steps to reproduce the issue

compile qgis 3.44.x on gentoo i.e. using 
```
emerge -1O qgis
```

### Versions

tried with 
- 3.44.3
- 3.44.4
- 3.44.5
- master at https://github.com/qgis/QGIS/commit/a1764d651f9913521fe319d94d50678686791f4f

### Supported QGIS version

- [x] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [ ] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

_No response_

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -618,7 +618,7 @@ if(WITH_CORE)
 
   find_package(${QT_VERSION_BASE} COMPONENTS Core Gui Widgets Network Xml Svg Concurrent Test Sql Positioning REQUIRED)
   if (BUILD_WITH_QT6)
-    find_package(${QT_VERSION_BASE} COMPONENTS Core5Compat REQUIRED)
+    find_package(${QT_VERSION_BASE} COMPONENTS Core5Compat SqlPrivate REQUIRED)
   endif()
 
   if (NOT IOS)

--- a/external/qspatialite/CMakeLists.txt
+++ b/external/qspatialite/CMakeLists.txt
@@ -5,6 +5,7 @@ add_definitions(${QT_DEFINITIONS})
 add_definitions(-DQT_PLUGIN)
 add_definitions(-DQT_SHARED)
 
+find_package(${QT_VERSION_BASE}Sql CONFIG)
 include_directories(SYSTEM ${${QT_VERSION_BASE}Sql_PRIVATE_INCLUDE_DIRS})
 
 set(QSQLSPATIALITE_SRC qsql_spatialite.cpp smain.cpp qsql_spatialite.h smain.h)

--- a/src/core/qgssnappingutils.cpp
+++ b/src/core/qgssnappingutils.cpp
@@ -646,7 +646,7 @@ QString QgsSnappingUtils::dump()
     msg += QString( "layer : %1\n"
                     "config: %2   tolerance %3 %4\n" )
            .arg( layer.layer->name() )
-           .arg( layer.type ).arg( layer.tolerance ).arg( static_cast<int>( layer.unit ) );
+           .arg( static_cast<int>( layer.type ) ).arg( layer.tolerance ).arg( static_cast<int>( layer.unit ) );
 
     if ( mStrategy == IndexAlwaysFull || mStrategy == IndexHybrid || mStrategy == IndexExtent )
     {

